aion/web/elm/src/Ranking/View.elm

```diff
@@ -2,56 +2,116 @@ module Ranking.View exposing (..)
                 [ rankingTable model ]
             ]
         ]
 
 
+selectCategoriesAttributes : Model -> List (Select.Item msg)
+selectCategoriesAttributes model =
+    case model.rankingData.data of
+        RemoteData.Success rankingData ->
+            List.map (\category -> Select.item [ value (toString category.categoryId) ] [ text category.categoryName] ) rankingData.rankingList
+        _ -> []
+
+
 rankingTable : Model -> Html Msg
 rankingTable model =
-    Table.simpleTable
-        ( Table.simpleThead
-            [ Table.th [] [ text "User" ]
+    Table.table
+        { options = [ Table.striped, Table.hover ]
+        , thead = Table.simpleThead
+            [ Table.th [] [ text "#" ]
+            , Table.th [ Table.cellAttr(style [ ("width", "5%") ]) ] []
+            , Table.th [] [ text "User" ]
             , Table.th [] [ text "Score" ]
             ]
-        , displayScores model.rankingData
-        )
+        , tbody = displayScores model
+        }
 
 
-displayScores : WebData RankingData -> Table.TBody Msg
-displayScores rankingData =
-    case rankingData of
+displayScores : Model -> Table.TBody Msg
+displayScores model =
+    case model.rankingData.data of
         RemoteData.Success rankingData ->
             let
-                sortedScores =
-                    List.reverse (List.sortBy .score rankingData.data)
+                categoryScores = selectedCategoryScores rankingData.rankingList model.rankingData.selectedCategoryId
             in
-                Table.tbody [] (List.map displaySingleScore sortedScores)
+                Table.tbody [] (List.map (displaySingleScore model) (sortedScoresWithIndices categoryScores))
 
         _ ->
             Table.tbody [] []
 
 
-displaySingleScore : PlayerScore -> Table.Row msg
-displaySingleScore score =
-    Table.tr []
-        [ Table.td [] [ text score.userName ]
-        , Table.td [] [ text (toString score.score) ]
-        ]
+displaySingleScore : Model -> (Int, PlayerScore) -> Table.Row msg
+displaySingleScore model indexAndScore =
+    case indexAndScore of
+        (index, playerScore) ->
+            Table.tr []
+                [ Table.td [] [ text (toString (index + 1)) ]
+                , Table.td [] (scoreImage model.location index)
+                , Table.td [] [ text playerScore.userName ]
+                , Table.td [] [ text (toString playerScore.score) ]
+                ]
+
+
+scoreImage : Navigation.Location -> Int -> List (Html msg)
+scoreImage location index =
+    let
+        commonPath = (host location) ++ "images/"
+        imagePath = case index of
+                        0 -> "medal_gold.png"
+                        1 -> "medal_silver.png"
+                        2 -> "medal_bronze.png"
+                        _ -> ""
+    in
+        case imagePath of
+            "" -> []
+            _ -> [ img [ src (commonPath ++ imagePath) ] [] ]
+
+
+selectedCategoryScores : List CategoryRanking -> Int -> List PlayerScore
```

Comment:
There's a lot of stuff happening here that doesn't really generate any html. I think it would be better to move them somewhere else (could be Model, could be a RankingUtils or whatever works best as a file for those).